src/docfx/lib/process/InterProcessMutex.cs

```diff
@@ -26,17 +26,25 @@ public static InterProcessMutex Create(string mutexName)
 
             var mutex = new Mutex(initiallyOwned: false, $"Global\\{HashUtility.GetMd5Hash(mutexName)}");
 
-            while (!mutex.WaitOne(TimeSpan.FromSeconds(30)))
+            try
             {
+                while (!mutex.WaitOne(TimeSpan.FromSeconds(30)))
+                {
 #pragma warning disable CA2002 // Do not lock on objects with weak identity
-                lock (Console.Out)
+                    lock (Console.Out)
 #pragma warning restore CA2002
-                {
-                    Console.ForegroundColor = ConsoleColor.Yellow;
-                    Console.WriteLine($"Waiting for another process to access '{mutexName}'");
-                    Console.ResetColor();
+                    {
+                        Console.ForegroundColor = ConsoleColor.Yellow;
+                        Console.WriteLine($"Waiting for another process to access '{mutexName}'");
+                        Console.ResetColor();
+                    }
                 }
             }
+            catch (AbandonedMutexException)
```

Comment:
This would only throw when the other thread exited without releasing its mutex. Why would it happen in the first place or did it happen during the testing?

Reply:
Like a process crash while holding a lock